node/src/actors/chain_manager/mining.rs

```diff
@@ -72,18 +72,14 @@ impl ChainManager {
             _ => return,
         };
 
-        if beacon.checkpoint > current_epoch {
+        if beacon.checkpoint >= current_epoch {
```

Comment:
Undocumented edit. What are you trying to conceal? :smirk:

Reply:
He's trying to introduce a backdoor :scream:

Reply:
Chain checkpoint beacon is the checkpoint beacon of the last consolidated block. Due to consolidate a block from epoch N is done in epoch N+1, our chain beacon always come from the past epoch. A chain beacon with the current epoch is the same error if it is come from the future

Reply:
Then put this change in a separate commit :P